internal/pkg/stream/cloudformation_test.go

```diff
@@ -240,6 +243,27 @@ func testStackStreamer_Fetch_WithError(t *testing.T) {
 	require.EqualError(t, err, "describe stack events phonetool-test: some error")
 }
 
+func testStackStreamer_Fetch_withThrottle(t *testing.T) {
+	// GIVEN
+	client := &mockCloudFormation{
+		err: awserr.New("RequestThrottled", "throttle err", errors.New("abc")),
+	}
+	streamer := &StackStreamer{
+		client:                *client,
+		stackName:             "phonetool-test",
+		changeSetCreationTime: time.Date(2020, time.November, 23, 16, 0, 0, 0, time.UTC),
+		pastEventIDs:          map[string]bool{},
+		retries:               0,
+	}
+
+	// WHEN
+	nextDate, err := streamer.Fetch()
+	maxDuration := 2 * streamerFetchIntervalDurationMs * time.Millisecond
+	require.NoError(t, err, "expect no results and no error for throttle exception")
+	require.True(t, nextDate.Before(time.Now().Add(maxDuration)), "expect that the returned timeout (%s) is less than the maximum for backoff (%d)", time.Until(nextDate), maxDuration)
```

Comment:
This is kindof annoying but I think to make the tests truly robust we will need to mock the current time. This file might help: https://github.com/aws/copilot-cli/blob/mainline/internal/pkg/term/progress/time.go

```go
type clock interface {
   now() time.Time
}

type realClock struct {}
func (c realClock) now() time.Time {
	return time.Now()
}

type fakeClock struct { now time.Time }
func (c fakeClock) now() time.Time {
	return c.now
}

type StackStreamer struct {
   ...
   clock clock // use a realClock in the constructor.
}
func nextFetchDate(clock clock, retries int) time.Time { ... }
```

Now in the tests we can create a `StackStreamer` with a `fakeClock` and measure exactly if the time difference is 4s.